src/main/cpp/subsystems/Drivetrain.cpp

```diff
@@ -406,6 +406,13 @@ bool Drivetrain::UpdateShuffleboard() {
             heading_pid_.SetPID(heading_p, heading_i, heading_d);
             turn_pid_.SetPID(turn_p, turn_i, turn_d);
         }
+
+        // Allow saving parameters in non-competition modes
+        if (DrivetrainUI::nt_save.GetBoolean(true)) {
+            // Save the parameters.
```

Comment:
Is there a way to get parameters off the robot (aside from maybe ssh)? To save them somewhere in the repo so they can be backed up here and so we can track changes?

Reply:
They can be FTP'd off the robot and saved in the repo. Shuffleboard itself has some sort of parameter management too if I remember correctly, but that should get overwritten by the robot's parameters.

Reply:
Also worth noting: every time we deploy to the robot the parameters get copied over. So any changes will probably be overwritten by deploy. We need to get better about our testing/tuning procedures to ensure our gains and parameters are not overwritten